src/libs/ReportUtils.js

```diff
@@ -3347,11 +3373,67 @@ function canAccessReport(report, policies, betas, allReportActions) {
  */
 function shouldHideReport(report, currentReportId) {
     const parentReport = getParentReport(getReport(currentReportId));
-    const reportActions = ReportActionsUtils.getAllReportActions(report.reportID);
-    const isChildReportHasComment = _.some(reportActions, (reportAction) => (reportAction.childVisibleActionCount || 0) > 0);
+    const allReportActions = ReportActionsUtils.getAllReportActions(report.reportID);
+    const isChildReportHasComment = _.some(allReportActions, (reportAction) => (reportAction.childVisibleActionCount || 0) > 0);
     return parentReport.reportID !== report.reportID && !isChildReportHasComment;
 }
 
+/**
+ * @param {String} transactionID
+ * @returns {Boolean}
+ */
+
+function transactionHasViolation(transactionID) {
+    const violations = lodashGet(transactionViolations, transactionID, []);
+    return _.some(violations, (violation) => violation.type === 'violation');
+}
+
+/**
+ *
+ * @param {Object} report
+ * @param {Array<String> | null} betas
+ * @returns {Boolean}
+ */
+
+function transactionThreadHasViolations(report, betas) {
+    if (!Permissions.canUseViolations(betas)) {
```

Comment:
@cead22 I've been fighting with the jest tests and it seems that I've fixed most of the issues, however the last one is that `SidebarOrderTest` seems to error because of a memory leak? [Seen here](https://github.com/Expensify/App/actions/runs/6961460233/job/18943088931?pr=31448). I've pinpointed that this check is causing it, but I've not been able to understand why it's causing this failure. Do you have any thoughts?

Reply:
Was able to resolve this! I was providing a default of object instead of an array in a certain spot.